src/ingest-pipeline/airflow/dags/imc_3d_deepcell_3dcc.py

```diff
@@ -0,0 +1,531 @@
+from airflow.operators.python import BranchPythonOperator, PythonOperator
+
+import utils
+from utils import (
+    get_cwltool_base_cmd,
+    get_dataset_uuid,
+    get_named_absolute_workflows,
+    get_parent_dataset_uuids_list,
+    get_parent_data_dir,
+    build_dataset_name as inner_build_dataset_name,
+    get_previous_revision_uuid,
+    get_uuid_for_error,
+    join_quote_command_str,
+    make_send_status_msg_function,
+    get_tmp_dir_path,
+    HMDAG,
+    get_queue_resource,
+    get_preserve_scratch_resource,
+)
+from hubmap_operators.common_operators import (
+    CleanupTmpDirOperator,
+    CreateTmpDirOperator,
+    JoinOperator,
+    LogInfoOperator,
+    MoveDataOperator,
+    SetDatasetProcessingOperator,
+)
+
+
+default_args = {
+    "owner": "hubmap",
+    "depends_on_past": False,
+    "start_date": datetime(2019, 1, 1),
+    "email": ["joel.welling@gmail.com"],
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 1,
+    "retry_delay": timedelta(minutes=1),
+    "xcom_push": True,
+    "queue": get_queue_resource("imc_3d_deepcell_3dcc"),
+    "on_failure_callback": utils.create_dataset_state_error_callback(get_uuid_for_error),
+}
+
+with HMDAG(
+    "imc_3d_deepcell_3dcc",
+    schedule_interval=None,
+    is_paused_upon_creation=False,
+    default_args=default_args,
+    user_defined_macros={
+        "tmp_dir_path": get_tmp_dir_path,
+        "preserve_scratch": get_preserve_scratch_resource("imc_3d_deepcell_3dcc"),
+    },
+) as dag:
+
+    pipeline_name = "imc-3d-segmentation"
+    cwl_workflows = get_named_absolute_workflows(
+        segmentation=Path(pipeline_name, "pipeline.cwl"),
+        sprm=Path("sprm", "pipeline.cwl"),
+        create_vis_symlink_archive=Path("create-vis-symlink-archive", "pipeline.cwl"),
+        ome_tiff_pyramid=Path("ome-tiff-pyramid", "pipeline.cwl"),
+        ome_tiff_offsets=Path("portal-containers", "ome-tiff-offsets.cwl"),
+        sprm_to_json=Path("portal-containers", "sprm-to-json.cwl"),
+        sprm_to_anndata=Path("portal-containers", "sprm-to-anndata.cwl"),
+    )
+
+    def build_dataset_name(**kwargs):
+        return inner_build_dataset_name(dag.dag_id, pipeline_name, **kwargs)
+
+    prepare_cwl_segmentation = DummyOperator(task_id="prepare_cwl_segmentation")
+
+    def build_cwltool_cwl_segmentation(**kwargs):
+        run_id = kwargs["run_id"]
+        tmpdir = get_tmp_dir_path(run_id)
+        print("tmpdir: ", tmpdir)
+        data_dir = get_parent_data_dir(**kwargs)
+        print("data_dir: ", data_dir)
+
+        workflow = cwl_workflows["segmentation"]
+        meta_yml_path = workflow.parent / "meta.yaml"
+
+        command = [
+            *get_cwltool_base_cmd(tmpdir),
+            # '--singularity',
+            workflow,
+            "--gpus=all",
+            "--cytoplasm_markers",
+            "In115",
+            "--membrane_markers",
+            "La139" "--nucleus_markers",
+            "Ir191" "--input_dir",
```

Comment:
Probably missing ',' after La139 and Ir191?

Reply:
Fixed; thanks!